Code/cmake/Modules/RDKitUtils.cmake

```diff
@@ -115,6 +115,10 @@ macro(rdkit_python_extension)
     PYTHON_ADD_MODULE(${RDKPY_NAME} ${RDKPY_SOURCES})
     set_target_properties(${RDKPY_NAME} PROPERTIES PREFIX "")
 
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      # Stop a giant spew of warnings with gcc and boost up to at least 1.71
+      target_compile_options(${RDKPY_NAME} PRIVATE -Wno-maybe-uninitialized)
+    endif()
```

Comment:
I haven't seen too may of these warnings... (actually, just 4, and not from boost... I addressed them with other warnings in a PR yesterday).

Anyway, I'm not really happy about passing that flag -- at some point we might start having uninitialized variables in our code, and we would miss them.

Reply:
If it’s boost only we can disable the warning in the booststartheaders include file:

https://github.com/rdkit/rdkit/blob/master/Code/RDGeneral/BoostStartInclude.h

We just need to turn them back on in boostendinclude

I’m starting some builds tonight.

Reply:
I agree that it's generally a bad idea to do this, but there are a huge number of those coming out of any file that include python_streambuf.h. I'm seeing this currently with boost 1.71, but I think it shows up earlier versions too.

Here's one of the ~20 that appear when building rdBase:
```
/opt/boost_1_71_0/include/boost/python/extract.hpp: In static member function ‘static bool boost::python::indexing_suite<Container, DerivedPolicies, NoProxy, NoSlice, Data, Index, Key>::base_contains(Container&, PyObject*) [with Container = std::__cxx11::list<int, std::allocator<int> >; DerivedPolicies = boost::python::detail::final_list_derived_policies<std::__cxx11::list<int, std::allocator<int> >, false>; bool NoProxy = false; bool NoSlice = false; Data = int; Index = long unsigned int; Key = int]’:
/opt/boost_1_71_0/include/boost/python/extract.hpp:185:11: warning: ‘*((void*)& x +24)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  185 |           );
      |           ^
```

@bp-kelley : addressing it by using BoostStartInclude.h with python_streambuf.h makes sense. I will take a look at that.